common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -2461,7 +2461,17 @@ public static enum ConfVars {
         "The default input format for tez. Tez groups splits in the AM."),
 
     HIVETEZCONTAINERSIZE("hive.tez.container.size", -1,
-        "By default Tez will spawn containers of the size of a mapper. This can be used to overwrite."),
+        "The memory in MB that's used by a Tez task container (TezChild) in Tez container mode. Hive uses this \n"
+        + "property to create a Resource object which is accepted by Yarn (and used in TezAM to ask for TezChild \n"
+        + "containers. This should be distinguished from the Tez AM's (DAGAppMaster) memory, \n"
+        + "which is driven by tez.am.resource.memory.mb! \n"
+        + "Also, as Hive takes care of TezChild memory by setting this option, there is no need \n "
+        + "to set tez.task.resource.memory.mb differently. \n"
+        + "One more thing to note that the final -Xmx arg for TezChild process is not equal to this setting, \n "
+        + "because Tez considers a heap fraction (80%), so by default: \n"
+        + "Xmx = hive.tez.container.size * tez.container.max.java.heap.fraction. \n"
+        + "In case of values <= 0, container size falls back to mapreduce.map.memory.mb. \n"
+        + "LLAP note: this option has no effect in LLAP mode, because LLAP deamons are sized in another way."),
```

Comment:
Based on the latest findings I am not sure if the `LLAP note` is still accurate. Should we remove it or mention the effect on split generation?

Reply:
1. no closing bracket after ` (and use`
2. could we drop everything after `One more thing to note that` and maybe add a link to https://blog.cloudera.com/optimizing-hive-on-tez-performance/

Reply:
1. I'll fix

2. maybe, but linking a cloudera blog post is not the Apache way, so if we want to do it, we need to have an Apache page I believe...regarding the content here: having 2-3 extra clarifying sentences should not hurt, because it clarifies the most common confusing stuff here immediately, I still think it's useful

Reply:
agreed, "One more thing to note that" is blog style, kind of useless in this context

Reply:
tnx!

Reply:
done